src/handler/mod.rs

```diff
@@ -0,0 +1,122 @@
+use anyhow::{bail, Error, Result};
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::process::Command;
+use std::str;
+
+pub fn handle_reboot(force: bool) -> Result<(), Error> {
+    if !force {
+        match get_boot_counter() {
+            Some(t) if t <= 0 => bail!("countdown ended, check greenboot-rollback status"),
+            None => bail!("boot_counter is not set"),
+            _ => {}
+        }
+    }
+    log::info!("restarting system");
+    Command::new("systemctl").arg("reboot").status()?;
```

Comment:
This should also check for `status.success()`...unfortunately Rust's std doesn't over a convenience helper for that.